### DISCUSSION TITLE
autopilot hitting NPCs

### DISCUSSION THREAD

**Discussion by sandykalugdan** [carla-simulator/carla#4781] [General]
Got this strange issue running my client...

My ego car is running in autopilot but for some reason, it is hitting NPCs in the simulation.  Using map Town01 and running on behaviour agent and v0.9.12 

https://user-images.githubusercontent.com/45012803/138557674-4dc50165-3929-4974-b31f-704c91e6b76d.mp4

**Comment by glopezdiest** [comment]
Hey @sandykalugdan, as for 0.9.12, agents are quite bad at detecting vehicle if either of them are part of an intersection. This was a known issue but we couldn't fix it until after releasing 0.9.12, so it's not working there. It is [these changes](https://github.com/carla-simulator/carla/pull/4613/files) plus [this minor fix](https://github.com/carla-simulator/carla/commit/4a32bfecb6addc377d2bd7b6510565a16212264d) for the basic agent

**Reply by sandykalugdan** [reply]
@glopezdiest , thank you for your response.   I'm going to try those fixes.

**Reply by ymurzello** [reply]
HI @glopezdiest 
I tried these fixes as well as the agent scripts in v0.9.13 and I'm still having the same issue of the autopilot hitting the npcs. I'm running this in Town01 but I think it's an issue in other Towns with junctions too.

**Comment by sandykalugdan** [comment]
Hello @glopezdiest, unfortunately, the codes didn't help much in solving my issue.  Using the provided fix, my client went totally uncontrollable and won't event activate my autopilot scenario.

**Reply by glopezdiest** [reply]
What do you mean by uncontrollable ? The code just alters the vehicle detection, so it shouldn't affect anything else

**Comment by iammix** [comment]
Hello @sandykalugdan
is your traffic manager correctly implemented? Autopilot works only with traffic manager.

**Reply by sandykalugdan** [reply]
Hello @iammix , 
Thanks for looking into this.
Yes, my traffic manager works fine when I use the basic_agent.py (without the fix).  When I use the version suggested, my ego vehicle went uncontrollable...

**Reply by iammix** [reply]
If you try to run the automatic_control.py file (with the behavior agent, not the basic) from the CARLA examples folder the same problem occurs or not?

**Reply by sandykalugdan** [reply]
Didn't get to that part yet but I'll so some tests tonight.  Isn't the behavior agent is now basic_agent in 0.9.12?

**Reply by iammix** [reply]
No, in 0.9.12 the basic_agent is different than the behavior_agent.
You can check that on your own at the following directory: 
[PythonAPI/carla/agents/navigation](https://github.com/carla-simulator/carla/tree/0.9.12/PythonAPI/carla/agents/navigation)

**Reply by sandykalugdan** [reply]
Thank you, I'll check this...
